nep-9.mediawiki

```diff
@@ -0,0 +1,161 @@
+{| class="wikitable"
+! URI Key
+! NEO Transaction attribute
+! Description
+|-
+| contractHash
+| 0x00
+| Hash value of contract 
+|-
+| ecdh02, ecdh03
+|	0x02,0x03
+|	Public key for ECDH key exchange
+|-
+| script 
+|	0x20
+|	Additional validation of transactions
+|-
+| vote
+|	0x30
+| For voting
+|-
+| certUrl
+| 0x80 
+|	Url address of certificate 
+|-
+| descriptionUrl
+| 0x81
+|	Url address of description 
+|-
+| description
+| 0x90
+| Brief description 
+|-
+| hash1,hash2,...,hash15
+|	0xa1,0xa2,...,0xaf
+|	Used to store custom hash values
+|-
+| remark,remark2,...,remark15
+|	0xf0,0xf1,...,0xff
+|	Remarks
+|}
+
+====Native Assets====  
+{| class="wikitable"
+! Name
+! Hash
+|-
+| NEO 
+| c56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b 
+|-
+| GAS 
+| 602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7 
+|}
+
+====Examples====
+
+Begin transaction to specified address
+
+<code>neo:AeNkbJdiMx49kBStQdDih7BzfDwyTNVRfb</code>
+
+
+Begin transaction of unspecified amount of NEO
+
+<code>neo:AeNkbJdiMx49kBStQdDih7BzfDwyTNVRfb?assetID=c56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b</code>
+
+Begin transaction to specified address of 1 NEO
+
+<code>neo:AeNkbJdiMx49kBStQdDih7BzfDwyTNVRfb?assetID=c56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b&amount=1.0</code>
+
+Begin transaction to specified address of 1 NEO with transaction attribute description = "Hello"
+
+<code>neo:AeNkbJdiMx49kBStQdDih7BzfDwyTNVRfb?assetID= c56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b&amount=1.0&description=Hello</code>
+
+Begin transaction to specified address of 0.1 GAS. Put public key in ecdh02 attribute field to allow sender to encrypt using ECDH. Transaction attribute description = "Hello"
+
+<code>neo:AQc5mtFayAdoCK13BW1cGAzAHyo9SoUWe7?assetID= 602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7&amount=0.1&ecdh02=02ed53ad58c838435d4dd7a4b25c1eba01384c814ca53a539405434807afbb04b4&description=Hello</code>
+
+
+This should be sufficient to facilitate transfer of native NEO assets safely.
+---
+
+===Smart Contract Invocation URI===
+To invoke the smart contract, we need to specify a script hash and the operation of the smart contract that is being invoked as well as providing required parameters. This way an application can tailor the screen to match the invoking operation with pre-filled information. e.g. transfer NEP-5 token, voting, etc. what left for a user to do is to authorize/sign the transaction. It would have the following URI. 
+
+<code>neo:<ScriptHash>?operation=<operation>[&<parameterName>=<parameterValue>][&<TransactionAttributeKey>=<value>]</code>
+
+===NEP-5 Token Transfer URI===
+
+A NEP-5 token transfer is a smart contract invocation and would have the following URI. 
+
+<code>neo:<ScriptHash>?operation=transfer&from=<address>&to=<address>&amount=<amount>[?<TransactionAttributeKey>=<value>]</code>
```

Comment:
any leeway for optionals? For example, I could have a donate some ONT tokens to me link with the `from` address empty. I could even leave `amount` empty for the user to decide how much to donate. Might make it messy though...

Reply:
I believe `from` and `amount` fields could be optional. it's pretty much like doing a native transfer. I'll update it.